### PR TITLE
Add permalink and refactor navigate of pages

### PR DESCRIPTION
**Pull request by Cxristian**

### CODE REVIEW

File: src/main/java/core/IStrategySetter.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ * @version 0.0.1
+ */
+@FunctionalInterface
+public interface IStrategySetter {
+
+    /**
+     * Executes method for a Page or an entity.
+     */
+    void executeMethod();
+}
```
Comment (aburgosu) [inline src/main/java/core/IStrategySetter.java]: add line.

File: src/main/java/orangeHRM/keys/CourseKeys.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    public static final String COST = "Cost";
+    public static final String COMPANY = "Company";
+    public static final String DURATION = "Duration";
+    public static final String DESCRIPTION = "Description";
+
+    /**
+     * Constructor of class.
+     */
+    private CourseKeys() { }
+}
```
Comment (aburgosu) [inline src/main/java/orangeHRM/keys/CourseKeys.java]: add white line

File: src/main/java/orangeHRM/ui/Permalink.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+import java.util.Map;
+
+/**
+ * Permalink class.
+ *
+ * @author Cristian Lujan
+ * @version 0.0.1
+ */
+public final class Permalink {
+
```
Comment (aburgosu) [inline src/main/java/orangeHRM/ui/Permalink.java:25]: delete white line

File: src/main/java/orangeHRM/ui/Permalink.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    public static String getPermalink(final String urlKey) {
+            return course.get(urlKey);
+    }
+
+    /**
+     * Constructor private.
+     */
+    private Permalink() { }
+
+}
```
Comment (aburgosu) [inline src/main/java/orangeHRM/ui/Permalink.java]: add white line

File: src/main/java/orangeHRM/ui/pages/CourseForm.java [hunk 5]
```diff
@@ -103,11 +118,4 @@ public void clickSaveBtn() {
...
         return confirmationMessage.getText();
     }
-
-    public CourseForm createCourse(final Courses course) {
-        setTitle(course.getTitle());
-        setCoordinator(course.getCoordinator());
-        clickSaveBtn();
-        return new CourseForm();
-    }
 }
```
Comment (aburgosu) [inline src/main/java/orangeHRM/ui/pages/CourseForm.java]: add white line

File: src/main/java/orangeHRM/ui/pages/CourseFormAbstract.java [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+    protected abstract void setCoordinator(String coordinator);
+
+    /**
+     * Sets the subunit sending a string.
+     *
+     * @param subunit for the contact.
+     */
+    protected abstract void setSubunit(String subunit);
+
+}
```
Comment (aburgosu) [inline src/main/java/orangeHRM/ui/pages/CourseFormAbstract.java]: Add white line

File: src/test/java/steps/CourseStep.java [hunk 7]
```diff
@@ -12,18 +12,18 @@
...
 import orangeHRM.entities.Context;
 import orangeHRM.entities.Courses;
 import orangeHRM.ui.pages.CourseForm;
+import orangeHRM.ui.pages.CourseFormAbstract;
 import orangeHRM.ui.pages.CoursePage;
 
 import java.util.Map;
 
-import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.*;
```
Comment (aburgosu) [inline src/test/java/steps/CourseStep.java]: import the needed classes only

File: src/test/resources/features/Course.feature [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    When I go to the COURSE_PAGE
+      And I open Course form
+      And I create a new Course with the following information in Course form
+        | Title       | Docker2.0     |
+        | Coordinator | Muhammad Khan |
+    Then a message that indicates the Course was created should be displayed
+    When I go to the COURSE_PAGE
+    Then the Course title should be displayed in the course list in the Course page
+
+#  Scenario: Create a new Course with all data
```
Comment (aburgosu) [inline src/test/resources/features/Course.feature]: try to delete all code commented

File: src/main/java/core/IStrategySetter.java (hunk 1 above)
Comment (Cxristian) [reply src/main/java/core/IStrategySetter.java]: ok

File: src/main/java/orangeHRM/keys/CourseKeys.java (hunk 2 above)
Comment (Cxristian) [reply src/main/java/orangeHRM/keys/CourseKeys.java]: ok

File: src/main/java/orangeHRM/ui/Permalink.java (hunk 3 above)
Comment (Cxristian) [reply src/main/java/orangeHRM/ui/Permalink.java:25]: ok

File: src/main/java/orangeHRM/ui/Permalink.java (hunk 4 above)
Comment (Cxristian) [reply src/main/java/orangeHRM/ui/Permalink.java]: ok

File: src/main/java/orangeHRM/ui/pages/CourseForm.java (hunk 5 above)
Comment (Cxristian) [reply src/main/java/orangeHRM/ui/pages/CourseForm.java]: ok

File: src/main/java/orangeHRM/ui/pages/CourseFormAbstract.java (hunk 6 above)
Comment (Cxristian) [reply src/main/java/orangeHRM/ui/pages/CourseFormAbstract.java]: ok

File: src/test/java/steps/CourseStep.java (hunk 7 above)
Comment (Cxristian) [reply src/test/java/steps/CourseStep.java]: ok

File: src/test/resources/features/Course.feature (hunk 8 above)
Comment (Cxristian) [reply src/test/resources/features/Course.feature]: ok

**Pull request merged**